canvas_site_creator/api.py

```diff
@@ -236,8 +238,19 @@ def create_canvas_course_and_section(request):
         course_sis_course_id=course_instance_id,
         course_term_id=term_id
     )
+
     try:
         course_result = create_new_course(**request_parameters).json()
+
+        # If this course is meant to be a blueprint course,
+        # the newly created course needs to have its blueprint field set to True
+        if is_blueprint:
+            update_parameters = dict(
+                request_ctx=SDK_CONTEXT,
+                id=course_result['id'],
+                course_blueprint=True
+            )
+            update_result = update_course(**update_parameters).json()
     except Exception as e:
         message = u'Error creating new course via SDK with request={}'.format(
             request_parameters)
```

Comment:
The exception block is for both the canvas calls.However the logging is relevant to the create call failing and prints the request_parameters even if there would be an exception with the second call. Maybe you could move the  update_course call into a separate block and log the update_parameters in the error message?

Reply:
I added a nested try except to log the update parameters and exception details if there is an error during the course update.